client/packages/invoices/src/InboundShipment/DetailView/Toolbar.tsx

```diff
@@ -49,6 +73,15 @@ export const Toolbar: FC = () => {
   ]);
   const { isGrouped, toggleIsGrouped } = useInbound.lines.rows();
   const t = useTranslation('replenishment');
+
+  const selectedIds = useSelectedIds();
+  const generateNewReturnLines = useReturns.lines.newReturnLines(selectedIds);
+
+  const onReturn = async () => {
+    const lines = await generateNewReturnLines();
+    if (lines) onReturnLines(lines);
+  };
```

Comment:
I'm not sure how I feel about giving the Toolbar the responsibility of querying for the new return lines, and only opening the modal if there are any.

Part of me wanted to do the querying for new return lines within the modal, but I don't like the idea of rendering the modal, only to determine there are no lines selected and having to close itself again?

Reply:
What about, instead of the Toolbar calling `onOpenReturns` directly, why not have it call a new function in the DetailView component that checks the selected lines and then opens the modal if appropriate? You could even have it so that it passes another prop `hasSelectedLines` to the Toolbar, and then have the Toolbar not even display the "Return" option unless its `true`?

Is it easy to get the selected state of all the lines within DetailView? (I would have thought so, but I'm not seeing it obviously available from a quick glance)